### PR TITLE
Mod and dependabot native

### PR DESCRIPTION
**Pull request by Nexushunter**
### Story:
Hello, fellow Workivians, I am opening this to assist with the transition to go mod. When attempting to install the frugal compiler in a Go modules environment the install will fail due to a lack of a mod file being present.

A Go Modules set up is:
- Go 1.15+
- Go 1.14 or below, with the `GOMODULE111` environment variable set to `on` or `auto`.

I have also transitioned your Dependabot config to use the Github Native version.

### Acceptance Criteria:
- [ ] At least one InfRe Squad 2 member has reviewed and +1'd
- [ ] Code has been tested and results documented
- [ ] Unit tests have been updated
- [ ] Updates to documentation if necessary
- [ ] Verify and document changes to any other Messaging components
- [ ] Pull request made against the 'develop' branch, not master

### Design Notes:
The change included affecting how the project handles its dependencies. It moves it to the official dependency manager of the language. The conversion of the Dependabot config was just a matter of updating them to the GH native counterparts.

### How To Test:
CI should pass, installing using mod should succeed, the Dependabot validator was successful.

### My Test Results:
Install using a Go Modules environment was successful.
frugal_install

#### Reviewers:
@Workiva/product2

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module github.com/Workiva/frugal
+
+go 1.15
```
Comment (Nexushunter) [inline go.mod:3]: I'm not sure if this will prevent consumption. But this can be brought down to `go 1.14` without issue.

**Review by natewoods-wf — APPROVED** [review]
This looks great! I've been looking forward to this PR since go modules were announced.  Nice work!

**Comment by natewoods-wf** [comment]
+10
```sh
$ pwd
/Users/natewoods/wk/frugal
$ git rev-parse HEAD
a16d8a8bb14128e00cb71a8acd23402dfbaec76b
$ sudo rm -rf ~/go/pkg  # nuke package cache
$ go1.15 run main.go --version
go: downloading github.com/urfave/cli v1.19.1
go: downloading golang.org/x/tools v0.0.0-20161215003254-dd796641777b
go: downloading gopkg.in/yaml.v2 v2.0.0-20160928153709-a5b47d31c556
frugal version 3.13.1
$ sudo rm -rf ~/go/pkg  # nuke package cache
$ go1.14 run main.go --version  # defaults to GO111MODULE=auto when not in GOPATH
go: downloading github.com/urfave/cli v1.19.1
go: downloading gopkg.in/yaml.v2 v2.0.0-20160928153709-a5b47d31c556
go: downloading golang.org/x/tools v0.0.0-20161215003254-dd796641777b
frugal version 3.13.1

```

**Review by danielharasymiw-wf — APPROVED** [review]

**Review by houstonking-wf — APPROVED** [review]

**Comment by natewoods-wf** [comment]
@Workiva/release-management-p for merge

**Pull request merged**